server/src/test/java/org/elasticsearch/index/mapper/NestedObjectMapperTests.java

```diff
@@ -742,4 +743,36 @@ public void testReorderParent() throws IOException {
         assertThat(doc.docs().get(1).get("nested1.field2"), equalTo("4"));
         assertThat(doc.docs().get(2).get("field"), equalTo("value"));
     }
+
+    public void testMergeNestedMappings() {
```

Comment:
It's great that you put together a real unit test. But for consistency with the rest of this class and how we currently test mapping behavior, we should perform these checks using higher-level services like `MapperService`. You can take a look at `testParentObjectMapperAreNested` in this class for a simple example. The test case `testLimitOfNestedFieldsPerIndex` shows how to merge in a new mapping.